### PR TITLE
Assets/responsive

### PR DESCRIPTION
**Pull request by mhewson**

### CODE REVIEW

File: src/components/Utilities/demos/visibility.vue [hunk 1]
```diff
@@ -14,16 +14,16 @@
       <thead>
         <tr>
           <th>class</th>
-          <th class="cdr-hide cdr-show@xs-only">
+          <th class="cdr-hide cdr-show@breakpoint--xs-only">
```
Comment (lancetharper) [inline src/components/Utilities/demos/visibility.vue]: did we decide to go with `@breakpoint` 🤢

Comment (mhewson) [reply src/components/Utilities/demos/visibility.vue]: Sure did - the thought was to be as explicit as possible

Comment (mhewson) [reply src/components/Utilities/demos/visibility.vue]: anyone want to see something else?

File: src/components/accordion/examples/Accordion.vue [hunk 2]
```diff
@@ -14,6 +14,7 @@
         Default
       </cdr-text>
       <cdr-accordion
+        spacing-class="cdr-space-mb-one-x@breakpoint--xs cdr-space-mb-four-x@breakpoint--sm"
```
Comment (lancetharper) [inline src/components/accordion/examples/Accordion.vue]: What is the point of this `spacing-class` prop all over? Why wouldn't we just use `class`?

File: src/components/Utilities/demos/visibility.vue (hunk 1 above)
Comment (lancetharper) [reply src/components/Utilities/demos/visibility.vue]: I thought that the `@` made it pretty clear already

Comment (KrisKnabel) [reply src/components/Utilities/demos/visibility.vue]: What are the other options besides '-only'? Is there also a '-greater' and '-less'?

Comment (KrisKnabel) [reply src/components/Utilities/demos/visibility.vue]: Is there also a 'greater' and 'less'?

File: src/components/accordion/examples/Accordion.vue (hunk 2 above)
Comment (mhewson) [reply src/components/accordion/examples/Accordion.vue]: My thinking is that we want to have a easy to understand API for the components. If every component has the same basic set of pops (size, space...) then you don't really need to think of how to do it - 
For spacing specifically from what I was seeing some times we want the spacing class to be applied in a nested component structure / composition -- this would not be possible with the native class name attr.

Comment (KrisKnabel) [reply src/components/accordion/examples/Accordion.vue]: What if we changed it from `spacing-class` to just `spacing`?  That would keep it consistent with `size`.  Alternatively we could just go with `space`.

File: src/components/Utilities/demos/visibility.vue (hunk 1 above)
Comment (lancetharper) [reply src/components/Utilities/demos/visibility.vue]: The breakpoints only go in one direction (up), `only` is just a convenience.

Want something large below `sm` but small above `md`? --> `large@xs small@md`

`large@xs` = large @ xs and up

which then gets changed `@md` by

`small@md` = small @ md and up

Comment (mhewson) [reply src/components/Utilities/demos/visibility.vue]: yea we talked about this at the last dev meeting and that thought was that we have other use-cases for xs sm md lg. The solution proposed was to make breakpoint explicit so there we no confusion  cdr-btn-sm@md vs cdr-btn-sm@breakpoint--md

Comment (cowills) [reply src/components/Utilities/demos/visibility.vue]: i think i was the one that brought up that "small@md" might get weird in variable names, but the variables arent exporting anything related to breakpoints it turns out >_<

i agree that "small@md" is better here

Comment (KrisKnabel) [reply src/components/Utilities/demos/visibility.vue]: I think that this is unfortunately something that consuming devs will have mixed feelings about.  Some prefer verbosity and other don't.  IMO I feel most developers at REI would prefer clarity so I'm kinda leaning towards Hewson's changes.

Comment (mhewson) [reply src/components/Utilities/demos/visibility.vue]: OK so you are all ok with the @sm ect?

Comment (KrisKnabel) [reply src/components/Utilities/demos/visibility.vue]: Should we create a poll in slack and send it to consumers?

Comment (mhewson) [reply src/components/Utilities/demos/visibility.vue]: :skull:

File: src/components/accordion/examples/Accordion.vue (hunk 2 above)
Comment (cowills) [reply src/components/accordion/examples/Accordion.vue]: presumably you could still just pass these directly into `class` if you preferred doing it that way, right?

i like the idea of having a way to group these classes logically, as it can be a bit weird using a dozen+ utility classes on something and having it just be one really long line of text under `class`. 

I dont have any opinions on the name other than it should be consistent (sizing/spacing, size/space, etc, etc,)

File: src/components/Utilities/demos/visibility.vue (hunk 1 above)
Comment (cowills) [reply src/components/Utilities/demos/visibility.vue]: isnt small@md what we're doing today? its kind of a breaking change to do it differently

File: src/components/Utilities/demos/inset.vue [hunk 3]
```diff
@@ -1,84 +1,25 @@
 <template>
-  <div data-backstop="inset-space-utilities">
+  <div data-backstop="inline-space-utilities">
```
Comment (KrisKnabel) [inline src/components/Utilities/demos/inset.vue]: It appeared above that `inline` was removed, should this actually be `inset`?

Comment (mhewson) [reply src/components/Utilities/demos/inset.vue]: ah yea

File: src/components/icon/examples/Icons.vue [hunk 4]
```diff
@@ -177,12 +177,6 @@ export default {
 <style>
   .icon-examples {
     line-height: 1;
-
-    .icon-hover {
-      &:hover {
-        fill: red;
-      }
-    }
   }
```
Comment (KrisKnabel) [inline src/components/icon/examples/Icons.vue:184]: This seems unrelated to Assets.  Is this intended?

**Review by KrisKnabel — DISMISSED** [review]

File: src/components/icon/examples/Icons.vue (hunk 4 above)
Comment (mhewson) [reply src/components/icon/examples/Icons.vue:184]: added that back in

**Review by KrisKnabel — APPROVED** [review]
👍
